data/transactions/logicsig.go

```diff
@@ -54,7 +54,12 @@ func (lsig *LogicSig) Len() int {
 	return lsiglen
 }
 
-// Equal returns true if both LogicSig are equivalent
+// Equal returns true if both LogicSig are equivalent.
+//
+// Out of paranoia, Equal distinguishes zero-length byte slices
+// from byte slice-typed nil values as they may have subtly
+// different behaviors within the evaluation of a LogicSig,
+// due to differences in msgpack encoding behavior.
 func (lsig *LogicSig) Equal(b *LogicSig) bool {
```

Comment:
Why does Equal take pointers as arguments? This will panic if either lsig or b is nil. (I'm not sure whether this is a problem now, but it might cause problems later)